staging/src/kubevirt.io/client-go/apis/snapshot/v1alpha1/types.go

```diff
@@ -203,3 +203,94 @@ type VolumeSnapshotStatus struct {
 	// +optional
 	Error *VirtualMachineSnapshotError `json:"error,omitempty"`
 }
+
+// VirtualMachineRestore defines the operation of restoring a VM
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type VirtualMachineRestore struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec VirtualMachineRestoreSpec `json:"spec"`
+
+	// +optional
+	Status *VirtualMachineRestoreStatus `json:"status,omitempty"`
+}
+
+// VirtualMachineRestoreSpec is the spec for a VirtualMachineRestoreresource
+type VirtualMachineRestoreSpec struct {
+	// initially only VirtualMachine type supported
+	Target corev1.TypedLocalObjectReference `json:"target"`
+
+	VirtualMachineSnapshotName string `json:"virtualMachineSnapshotName"`
+
+	// only one of IncludeVolumes/ExcludeVolumes may be specified
+	// +optional
+	IncludeVolumes []string `json:"includeVolumes,omitempty"`
+
+	// +optional
+	ExcludeVolumes []string `json:"excludeVolumes,omitempty"`
+}
+
+// VirtualMachineRestoreStatus is the spec for a VirtualMachineRestoreresource
+type VirtualMachineRestoreStatus struct {
+	// +optional
+	RestoreTime *metav1.Time `json:"restoreTime,omitempty"`
```

Comment:
Is this the time the operation has started or the time it has reached its final state?

Reply:
I was thinking it would get set at the same time as "complete" so final state